Core/Inc/processing.h

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file processing.h
+ * @author Scott Abramson
+ * @brief Functions and tasks for processing data.
+ * @version 0.1
+ * @date 2024-08-04
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#ifndef PROCESSING_H
+#define PROCESSING_H
+
+#include "cmsis_os.h"
+#include "stdbool.h"
+
+#define TSMS_UPDATE_FLAG 1U
+#define PEDAL_DATA_FLAG	 1U
+
+#define PIT_MAX_SPEED	 5.0 /* mph */
+#define ACCUMULATOR_SIZE 10 /* size of the accumulator for averaging */
+
+/*
+ * Increases the torque limit by 10%
+*/
+void increase_torque_limit();
+
+/*
+ * Decreases the torque limit by 10%
+*/
+void decrease_torque_limit();
+
+/**
+ * @brief Get the debounced TSMS reading.
+ * 
+ * @return State of TSMS.
+ */
+bool get_tsms();
+
+/**
+ * @brief Task for processing data from the TSMS.
+ */
+void vProcessTSMS(void *pv_params);
```

Comment:
Yes since we only want to run only the logic to process new TSMS data when there is new TSMS data. Thus, only the logic for TSMS processing will be run when the flag indicating new tsms data is set high.

Reply:
i dont really think this needs its own thread, a mutex or semaphore or interuupt can signal a larger thread to process when receieving tsms data. But ur call in the end